assets/javascripts/discourse/components/json-editor.js.es6

```diff
@@ -0,0 +1,14 @@
+import AceEditor from "admin/components/ace-editor";
+
+export default class JsonEditor extends AceEditor {
+  mode = "json";
+  htmlPlaceholder = true;
+
+  _overridePlaceholder(loadedAce) {
```

Comment:
@angusmcleod this is the simplest way I found to plug into the Ace editor configuration from the Discourse component (without fully copying it), though it feels quite "hacky" as you can see.

Although it is nice to have the JSON syntax highlighting, it is also an option to just use the plain version of the editor instead and not include the additional Ace related files.

Reply:
I'm happy with this extension of the core class. Slightly hacky, but not that bad in the scheme of things. You have to do this kind of thing sometimes in a plugin.